source/common/ssl/context_impl.cc

```diff
@@ -54,14 +54,33 @@ ContextImpl::ContextImpl(ContextManagerImpl& parent, Stats::Scope& scope,
 
   int verify_mode = SSL_VERIFY_NONE;
 
-  if (!config.caCertFile().empty()) {
-    ca_cert_ = loadCert(config.caCertFile());
-    ca_file_path_ = config.caCertFile();
-    // set CA certificate
-    int rc = SSL_CTX_load_verify_locations(ctx_.get(), config.caCertFile().c_str(), nullptr);
-    if (0 == rc) {
+  if (!config.caCert().empty()) {
+    ca_file_path_ = config.caCertPath();
+    bssl::UniquePtr<BIO> bio(
+        BIO_new_mem_buf(const_cast<char*>(config.caCert().data()), config.caCert().size()));
+    RELEASE_ASSERT(bio != nullptr);
+    // Based on BoringSSL's X509_load_cert_crl_file().
```

Comment:
Not familiar enough with BoringSSL to know if this is the best we can do. Is there no way to do a non-filesystem version of things like `SSL_CTX_load_verify_locations` other than reimplementing bits of BoringSSL? Seems like the kind of place tmpfs could be useful but this isn't portable or even easy to manager at user privilege levels. @davidben for thoughts as well.

Reply:
It's unfortunate that the openssl/boringssl has nice APIs for files, but not an analogous API for a BIO.  But since that seems to be the way it is for now, I think this is the best we can do.  I think a tmpfs is a huge can of worms to open up, with security and operational issues.

Reply:
Unfortunately, there are no non-filesystem equivalents of the APIs we use, and even if I could add them today, we're on the `chromium-stable` branch, which is ~2 months behind the `master` branch, so it wouldn't help anyway.

I have a mid-term plan to migrate away from `X509` APIs to BoringSSL-specific [`CRYPTO_BUFFERs`](https://boringssl.googlesource.com/boringssl/+/HEAD/PORTING.md#crypto_buffer) and associated APIs, but that's further down the line, and it requires custom certificate verification logic.